mentat/code_context.py

```diff
@@ -262,62 +206,26 @@ async def _get_code_message(
         code_message += ["Code Files:\n"]
         meta_tokens = count_tokens("\n".join(code_message), model)
 
-        # Add user-included features
-        included_features = self._get_include_features()
-        included_feature_tokens = sum(
-            await count_feature_tokens(included_features, model)
-        )
-        remaining_tokens = max(0, max_tokens - meta_tokens - included_feature_tokens)
         auto_tokens = config.auto_tokens
-        if remaining_tokens == 0 or config.auto_tokens == 0:
-            self.features = sorted(
-                included_features, key=lambda f: f.path.relative_to(git_root)
-            )
+        remaining_tokens = max(0, min(auto_tokens, max_tokens - meta_tokens))
+
+        if remaining_tokens == 0:
+            self.features = self._get_include_features()
         else:
-            auto_tokens = (
-                remaining_tokens
-                if auto_tokens is None
-                else min(remaining_tokens, auto_tokens)
+            self.features = self._get_all_features(
+                CodeMessageLevel.INTERVAL,
             )
-            selectable_tokens = included_feature_tokens + auto_tokens
 
-            # Get a complete, sorted list of features to select from
-            if prompt and config.use_embeddings:
-                candidate_features = await self.search(
-                    prompt, level=CodeMessageLevel.INTERVAL
-                )
-                candidate_features = [
-                    f[0] for f in candidate_features
-                ]  # Drop the score, keep sorted
-            else:
-                candidate_features = _get_all_features(
-                    git_root,
-                    self.include_files,
-                    self.ignore_files,
-                    self.diff_context,
-                    self.code_map,
-                    CodeMessageLevel.INTERVAL,
-                )
-            _user = [f for f in candidate_features if f.user_included]
-            _non = [f for f in candidate_features if not f.user_included]
-            candidate_features = _user + _non  # Move included files to the front
-
-            alt_levels = [CodeMessageLevel.FILE_NAME]
-            if self.code_map:
-                alt_levels = [
-                    CodeMessageLevel.CMAP_FULL,
-                    CodeMessageLevel.CMAP,
-                ] + alt_levels
-
-            feature_selector = get_feature_selector(self.use_llm)
-            self.features = await feature_selector.select(
-                candidate_features,
-                selectable_tokens,
-                model,
-                alt_levels,
-                prompt,
-                expected_edits,
-            )
+        feature_filter = DefaultFilter(
```

Comment:
I put flags in `DefaultFilter` and `EmbeddingSimilarityFIlter` (Running `/search ""` was also throwing an error). I moved this filter into the if statement for now. The reason I didn't originally is because I think we should move to a place where `mentat .` works and to do that we'll need filtering on the included features as well.